pkg/cluster/hive_test.go

```diff
@@ -0,0 +1,156 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_hive "github.com/Azure/ARO-RP/pkg/util/mocks/hive"
+	testdatabase "github.com/Azure/ARO-RP/test/database"
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+//Test that namespace is created if it doesn't exist
+func TestHiveCreateNamespaceShouldCreateNamespace(t *testing.T) {
```

Comment:
Could you please refactor these tests to be a single test function with subtests (e.g. each test case in `t.Run(...)`).

We have a lot of tests like this in the codebase if you want to use it as an example.